docs/src/man/basics.md

```diff
@@ -0,0 +1,2042 @@
+
+julia> transform(german, :Age => :Sex, :Sex => :Age) # swapping the value of `:Age` column with `:Sex` column
+1000×10 DataFrame
+  Row │ id     Age     Sex    Job    Housing  Saving accounts  Checking accoun ⋯
+      │ Int64  String  Int64  Int64  String   String           String          ⋯
+──────┼─────────────────────────────────────────────────────────────────────────
+    1 │     0  male       67      2  own      NA               little          ⋯
+    2 │     1  female     22      2  own      little           moderate
+    3 │     2  male       49      1  own      little           NA
+    4 │     3  male       45      2  free     little           little
+    5 │     4  male       53      2  free     little           little          ⋯
+    6 │     5  male       35      1  free     NA               NA
+    7 │     6  male       53      2  own      quite rich       NA
+    8 │     7  male       35      3  rent     little           moderate
+  ⋮   │   ⋮      ⋮       ⋮      ⋮       ⋮            ⋮                ⋮        ⋱
+  994 │   993  male       30      3  own      little           little          ⋯
+  995 │   994  male       50      2  own      NA               NA
+  996 │   995  female     31      1  own      little           NA
+  997 │   996  male       40      3  own      little           little
+  998 │   997  male       38      2  own      little           NA              ⋯
+  999 │   998  male       23      2  free     little           little
+ 1000 │   999  male       27      2  own      moderate         moderate
+                                                  4 columns and 985 rows omitted
+
+julia> german_copy = german[:, [:Age, :Job]] # getting two columns
+1000×2 DataFrame
+  Row │ Age    Job
+      │ Int64  Int64
+──────┼──────────────
+    1 │    67      2
+    2 │    22      2
+    3 │    49      1
+    4 │    45      2
+    5 │    53      2
+    6 │    35      1
+    7 │    53      2
+    8 │    35      3
+  ⋮   │   ⋮      ⋮
+  994 │    30      3
+  995 │    50      2
+  996 │    31      1
+  997 │    40      3
+  998 │    38      2
+  999 │    23      2
+ 1000 │    27      2
+     985 rows omitted
+
+julia> transform(german_copy, [:Age, :Job] => (+) => :res) # put the result of `:Age` and `:Job` column after addition in `:res` column
+1000×3 DataFrame
+  Row │ Age    Job    res
+      │ Int64  Int64  Int64
+──────┼─────────────────────
+    1 │    67      2     69
+    2 │    22      2     24
+    3 │    49      1     50
+    4 │    45      2     47
+    5 │    53      2     55
+    6 │    35      1     36
+    7 │    53      2     55
+    8 │    35      3     38
+  ⋮   │   ⋮      ⋮      ⋮
+  994 │    30      3     33
+  995 │    50      2     52
+  996 │    31      1     32
+  997 │    40      3     43
+  998 │    38      2     40
+  999 │    23      2     25
+ 1000 │    27      2     29
+            985 rows omitted
+
+julia> transform(groupby(german, :Sex),
```

Comment:
this example seems completely out of place here

Reply:
I was discussing `transform` operations so I thought to introduce an example on GroupedDataFrame but in my PR #2803 I have covered everything about *GroupedDataFrame*.

Reply:
PRs must be stand-alone as we have discussed. I would remove this example then, and add what needs to be added in #2803 later. The reason is that at this point the user does not even know what `groupby` does.